src/utils/colors.ts

```diff
@@ -118,20 +125,19 @@ export const defaultPalette = [
 
 /**
  * Generate colors from the official nextcloud color
- * You can provide how many colors you want (multiplied by 3)
- * if step = 6
- * 3 colors * 6 will result in 18 generated colors
+ * You can provide how many colors you want (multiplied by 3).
+ *
+ * For example if steps is set to 6, then 3 colors * 6 will result in 18 generated colors.
  *
- * @param {number} [steps] Number of steps to go from a color to another
- * @return {Color[]}
+ * @param steps - Number of steps to go from a color to another
  */
-export function GenColors(steps) {
-	if (!steps) {
+export function generatePalette(steps: number): Color[] {
+	if (steps <= 0) {
 		steps = 6
 	}
 
 	if (steps === 4) {
-		return defaultPalette
+		return [...defaultPalette]
```

Comment:
Why?
If to make it immutable, we need to clone Colors as well

Reply:
I do not think its a problem of immutable colors but of the wrapping array as this is on module scope.
But good point.

Reply:
> but of the wrapping array as this is on module scope.

Why is it a problem?

Reply:
Colors inside the array are also in the module-scope

Reply:
it should in general immutable IMHO